server/controllers/user.js

```diff
@@ -11,9 +49,14 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
   if (searchString) {
     users = await User.find({
       username: { $regex: searchString, $options: "i" },
-    });
+    })
+      .select(" -password  ")
+      .populate("profile", "firstName lastName")
+      .exec()
+      .then((result) => {
+        return result.filter((userWithProfile) => !!userWithProfile.profile);
+      });
```

Comment:
You could just use only `await` without this `then` since you're using an `async` function.

Reply:
done ✔️